### DISCUSSION TITLE
Run JS on page update

### DISCUSSION THREAD

**Discussion by jeroenvanrensen** [livewire/livewire#6392] [1. Help] [answered]
Hello everyone,

Every time Livewire updates the page (after a server request), I want to execute some JavaScript.

I can't however figure out which hook to use.

Some hooks I tried:

- `Livewire.hook('component.rendered')`
- `Livewire.hook('rendered')`
- `Livewire.on('rendered')`

Etc.

Thank you in advance!

**Comment by danie-ramdhani** [comment]
like this?

```php
    public function render()
    {
        $this->js(<<<'JS'
            alert('hello there!');
        JS);
        
        return view('livewire.my-view');
    }
```

**Reply by jeroenvanrensen** [reply]
Not really, I need to call the javascript code after the page has been modified/rendered by Livewire. When running `$this->js()` the javascript gets called before the dom updates.

**Comment by jeroenvanrensen** [comment, marked as answer]
I found the answer: wrap the function in a timeout call, so: `$this->js('setTimeout(renderMath)');`
